zcash_client_backend/src/fees/zip317.rs

```diff
@@ -425,14 +425,87 @@ mod tests {
                 result,
                 Ok(balance) if
                     balance.proposed_change() == [
-                        ChangeValue::sapling(Zatoshis::const_from_u64(124_7500), None),
-                        ChangeValue::sapling(Zatoshis::const_from_u64(124_2500), None),
-                        ChangeValue::sapling(Zatoshis::const_from_u64(124_2500), None),
-                        ChangeValue::sapling(Zatoshis::const_from_u64(124_2500), None),
+                        ChangeValue::sapling(Zatoshis::const_from_u64(124_3750), None),
```

Comment:
The altered logic now evenly splits the change instead of adding excess to the first change output.